utils/artifacts.py

```diff
@@ -53,10 +51,16 @@ def get_release_and_version(package_path: str) -> Tuple[bool, str, str, str, str
     """
     Load version and release info from deepsparse package
     """
+    # deepsparse/src/deepsparse/version.py always exists, default source of truth
+    version_path = os.path.join(package_path, "version.py")
 
-    version_path = os.path.join(package_path, "generated_version.py")
-    if not os.path.exists(version_path):
-        version_path = os.path.join(package_path, "version.py")
+    # If deepsparse/engine-version.txt exists, use the generated_version
+    engine_version_path = os.path.abspath(
+        os.path.join(package_path, os.pardir, os.pardir, "engine-version.txt")
```

Comment:
How does checking the presence of engine-version.txt break the cycle you mention in the PR description? Is it not possible to have an old engine-version.txt lying around?

Reply:
The key thing is `generated_version.py` is pulled down with the binaries you download, `engine-version.txt` is not. The only current way to get an `engine-version.txt` is if you are creating the binaries locally yourself. We don't want to disrupt this flow by always using the deepsparse repo as source of truth.
If we wanted to do the safest approach, I think that would be to compare the versions in `version.py` and `generated_version.py`, only using `version.py` if it is newer. It'll be a bit messier but should work